src/crypto/tls/delegated_credentials.go

```diff
@@ -296,20 +303,22 @@ func getSignatureAlgorithm(cert *Certificate) (SignatureScheme, error) {
 	case *ecdsa.PrivateKey:
 		pk := sk.Public().(*ecdsa.PublicKey)
 		curveName := pk.Curve.Params().Name
-		certAlg := cert.Leaf.SignatureAlgorithm
-		if certAlg == x509.ECDSAWithSHA256 && curveName == "P-256" {
+		certAlg := cert.Leaf.PublicKeyAlgorithm
```

Comment:
GREAT catch. It seems like a huge oversight that the distinction between SignatureAlgorithm and PublicKeyAlgorithm is not explicitly documented. This seems easy to confuse (as we did!)